istioctl/cmd/analyze.go

```diff
@@ -109,6 +112,9 @@ istioctl analyze -d true a.yaml b.yaml services.yaml
 # Analyze the current live cluster, overriding service discovery to disabled
 istioctl analyze -k -d false
 
+# Analyze the current live cluster and suppress PodMissingProxy for pod mypod in namespace 'testing'.
```

Comment:
It would be useful to be able to suppress a message for all resources.

Reply:
Agree - I'm not sure it's in the document or not but we want to support basic globbing as well. I can include it in this PR if you feel strongly about it.

Reply:
IMO globbing can wait, but it's probably a good idea (and hopefully pretty straightforward?) to handle the code-only case. I don't mind this getting pushed to a fast-followup PR, but my impression is that it's simple enough and coupled enough that future PR readers will have an easier time if you just add it now.

Reply:
I went ahead and added it now. Turned out to be easy as go already supports it.

It leverages go's filepath globbing logic (https://golang.org/pkg/path/filepath/#Match), so supported are *, ?, and basic character classes.

Note that globbing isn't supported on the message code (I couldn't figure out a use case, there). Also note that the pattern that the user writes is directly passed to `filepath.Match` without any validation/checking. If the pattern is bad, it's silently ignored. I think this is safe to do as the language is so limited (a general regexp might not be safe to do).